x/blob/ante/ante.go

```diff
@@ -0,0 +1,65 @@
+package ante
+
+import (
+	"github.com/celestiaorg/celestia-app/pkg/appconsts"
+	"github.com/celestiaorg/celestia-app/pkg/shares"
+	"github.com/celestiaorg/celestia-app/x/blob/types"
+
+	"cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+// MinGasPFBDecorator helps to prevent a PFB from being included in a block
+// but running out of gas in DeliverTx (effectively getting DA for free)
+// This decorator should be run after any decorator that consumes gas.
+type MinGasPFBDecorator struct {
+	k BlobKeeper
+}
+
+func NewMinGasPFBDecorator(k BlobKeeper) MinGasPFBDecorator {
+	return MinGasPFBDecorator{k}
+}
+
+// AnteHandle implemnts the AnteHandler interface. It checks to see
+// if the transaction contains a MsgPayForBlobs and if so, checks that
+// the transaction has allocated enough gas.
+//
+// TODO: We need to run this antehandler in process proposal
+// hoever this will be a breaking change.
```

Comment:
[nit]

```suggestion
// however this will be a breaking change.
```

We should consider additional linters to catch typos like this. I would've expected https://github.com/celestiaorg/celestia-app/blob/c8f5c3376d6d8da214bd3af677c4d23970ea807b/.golangci.yml#L9 to catch 

Ref: https://golangci-lint.run/usage/linters/#misspell